File: leetcode/fun with ararys 101/findCountofEvenLengthNumbers.c
// Find Numbers with Even Number of Digits

// Given an array nums of integers, return how many of them contain an even number of digits.
int findNumbers(int* nums, int numsSize) {
    int count  = 0;
    // MY FIRST APPROACH BUT IT IS TAKING MORE CPU TIME
    // for(int i = 0; i < numsSize; i++) {
    //     char convert_string[1000000];
    //     sprintf(convert_string, "%d",nums[i]);
    //     if(strlen(convert_string)%2 == 0) {
    //         count++;
    //     }
    // }
    
    //OPTIMIZED
    for(int i = 0; i< numsSize; i++){
        int num = nums[i];
        int quotient_count = 0;
        while(num > 0){
            num = num /10;
            quotient_count++;
        }
        if( quotient_count % 2 == 0){
            count++;
        }
    }
    return count;
}

File: leetcode/fun with ararys 101/findCountofEvenLengthNumbers.py
# Find Numbers with Even Number of Digits

# Given an array nums of integers, return how many of them contain an even number of digits.
count = 0
nums = [12,345,2,6,789]
for i in nums:
    float_div_val = i/2
    if i > 9 and float_div_val == int(float_div_val): # same as i//2
        count = count + 1
print(count)